src/test/java/seedu/guest/model/guest/DateRangeTest.java

```diff
@@ -0,0 +1,43 @@
+package seedu.guest.model.guest;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.guest.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class DateRangeTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new DateRange(null));
+    }
+
+    @Test
+    public void constructor_invalidDateRange_throwsIllegalArgumentException() {
+        String invalidDateRange = "";
+        assertThrows(IllegalArgumentException.class, () -> new DateRange(invalidDateRange));
+    }
+
+    @Test
+    public void isValidDateRange() {
+        // null date range
+        assertThrows(NullPointerException.class, () -> DateRange.isValidDateRange(null));
+
+        // invalid date range
+        assertFalse(DateRange.isValidDateRange("")); // empty string
+        assertFalse(DateRange.isValidDateRange("13/09/22")); // 1 date only
+        assertFalse(DateRange.isValidDateRange("13/09/22 15/09/22")); // wrong separator
+        assertFalse(DateRange.isValidDateRange("13/09/2022 - 15/09/2022")); // wrong date format
+        assertFalse(DateRange.isValidDateRange("13/9/22 - 15/9/22")); // wrong date format
```

Comment:
Should we check wrong date format for the days as well? eg 1/09/22 - 3/09/22

Reply:
Yes, added and included checking of year format